src/helper/ssg/utils.test.ts

```diff
@@ -0,0 +1,19 @@
+import { describe, it, expect } from 'vitest'
+import { joinPaths, dirname } from './utils'
+
+describe('joinPath', () => {
+  it('Should joined path is valid.', () => {
+    expect(joinPaths('test', 'test2')).toBe('test/test2')
+    expect(joinPaths('test', 'test2', '../test3')).toBe('test/test3')
+    expect(joinPaths('.', '../')).toBe('..')
```

Comment:
Could you add these cases to cover all patterns?

```ts
expect(joinPaths('test/', 'test2/')).toBe('test/test2') // trailing slashes
expect(joinPaths('./test', './test2')).toBe('test/test2') // dot and slash
expect(joinPaths('', 'test')).toBe('test') // empty path
expect(joinPaths('/test', '/test2')).toBe('test/test2') // root path
```

Reply:
I further believe that the following cases are also necessary.
```typescript
expect(joinPaths('../', 'test')).toBe('../test') // parent and single
```

Reply:
@yusukebe @EdamAme-x I added those test code, and passed test.

@yusukebe
```ts
expect(joinPaths('/test', '/test2')).toBe('test/test2') // root path
```
Is this code valid?
It test is successful, but I think `joinPaths('/test', '/test2')` should be `/test2`

Reply:
> Is this code valid?
> It test is successful, but I think joinPaths('/test', '/test2') should be /test2

I tried this code using `node:path`, the result is `/test/test2`. So I will fix code.

Reply:
I fixed.